idioms/rustdoc-init.md

```diff
@@ -0,0 +1,73 @@
+# Easy doc initialization
+
+## Description
+
+If a struct takes significant effort to initialize, when writing docs it can be quicker to wrap your example with a 
+function the struct as an argument.
+
+## Motivation
+Sometimes there is a struct with multiple or complicated parameters and several methods.
+Each of these methods should have examples. 
+
+For example:
+
+```rust
+struct ExampleStruct {
+    foo: Box<Foo>,
+    count: AtomicI64,
+}
```

Comment:
Can we use some more realistic example rather some weird example here?

Reply:
Done

Reply:
Oh, nice example.